TestsReportGenerator: Always show the error message, and truncate only the stdout

radical opened the pull request:
The combined error message and stdout were being truncated, which meant that the former would not show, like:

<img width="1792" alt="Screenshot 2025-05-15 at 14 31 58" src="https://github.com/user-attachments/assets/f53c5232-b05b-4f02-88e3-af3cc8a16472" />

Also:
- Quarantine Aspire.Hosting.Tests.DistributedApplicationTests.TestServicesWithMultipleReplicas - https://github.com/dotnet/aspire/issues/9340
- Quarantine YarpFunctionalTests.VerifyYarpResource - https://github.com/dotnet/aspire/issues/9344

radical committed:
TestsReportGenerator: Always show the error message, and truncate only the stdout

danmoseley commented on tools/GenerateTestSummary/TestSummaryGenerator.cs:
````diff
@@ -144,21 +144,17 @@ public static void CreateSingleTestSummaryReport(string trxFilePath, StringBuild
-                var errorMsgBuilder = new StringBuilder();
-                errorMsgBuilder.AppendLine(test.Output?.ErrorInfo?.InnerText ?? string.Empty);
+                reportBuilder.AppendLine();
+                reportBuilder.AppendLine("```yml");
````
yml? do you mean text maybe?

radical replied:
That seems to provide a nice coloring!

danmoseley commented on tools/GenerateTestSummary/TestSummaryGenerator.cs:
````diff
@@ -144,21 +144,17 @@ public static void CreateSingleTestSummaryReport(string trxFilePath, StringBuild
+                reportBuilder.AppendLine();
+                reportBuilder.AppendLine("```yml");
+
+                reportBuilder.AppendLine(test.Output?.ErrorInfo?.InnerText ?? string.Empty);
````
```suggestion
                reportBuilder.AppendLine(test.Output?.ErrorInfo?.InnerText);
```
null is blank

radical committed:
Update tools/GenerateTestSummary/TestSummaryGenerator.cs

Co-authored-by: Dan Moseley <danmose@microsoft.com>

danmoseley commented on tools/GenerateTestSummary/TestSummaryGenerator.cs:
```diff
@@ -144,21 +144,17 @@ public static void CreateSingleTestSummaryReport(string trxFilePath, StringBuild
-                    errorMsgBuilder.AppendLine(test.Output.StdOut);
+                    reportBuilder.AppendLine();
+                    reportBuilder.AppendLine("### StdOut");
+                    reportBuilder.AppendLine(TruncateTheStart(test.Output.StdOut, 50_000));
```
could it be that the meat is ever at the end? ie would be better to take the first 50_000, then the last (length - 50_000 > 0 ? 100 : 0) or something

idk

radical replied:
My thinking was that we might see failed resources or exceptions towards the end. And the beginning would be more of starting up messages. Though the end also last few lines of shutdown messages.

radical replied:
I don't have a strong opinion on this though.

danmoseley replied:
I mistyped, I meant what about taking the first 25_000, the last 25_000, and removing any overlap. So all you lose is in the middle

radical replied:
Done.

radical replied:
This is what it will look like now:
```
| [2025-05-15T16:36:23] Aspire.Hosting.Tests.Resources.backend Information: 29: 2025-05-15T16:36:23.8940000Z 14da29b7e816: Download complete
| [2025-05-15T16:36:23] Aspire.Hosting.Tests.Resources.backend Information: 30: 2025-05-15T16

... (snip) ...

... sources.yarp Information: 50: 2025-05-15T16:36:28.1672245Z dbug: Microsoft.Extensions.ServiceDiscovery.ServiceEndpointWatcher[1]
| [2025-05-15T16:36:28] Aspire.Hosting.Tests.Resources.yarp Information: 51: 2025-05-15T16:36:28.1672374Z       Resolving endpoints for service 'http://backend'.
```

radical committed:
Address review feedback from @ danmoseley .

feedback: `what about taking the first 25_000, the last 25_000, and removing any overlap. So all you lose is in the middle`

```
| [2025-05-15T16:36:23] Aspire.Hosting.Tests.Resources.backend Information: 29: 2025-05-15T16:36:23.8940000Z 14da29b7e816: Download complete
| [2025-05-15T16:36:23] Aspire.Hosting.Tests.Resources.backend Information: 30: 2025-05-15T16

... (snip) ...

... sources.yarp Information: 50: 2025-05-15T16:36:28.1672245Z dbug: Microsoft.Extensions.ServiceDiscovery.ServiceEndpointWatcher[1]
| [2025-05-15T16:36:28] Aspire.Hosting.Tests.Resources.yarp Information: 51: 2025-05-15T16:36:28.1672374Z       Resolving endpoints for service 'http://backend'.
```

radical committed:
Quarantine Aspire.Hosting.Tests.DistributedApplicationTests.TestServicesWithMultipleReplicas

Issue: https://github.com/dotnet/aspire/issues/9340

radical committed:
Quarantine YarpFunctionalTests.VerifyYarpResource - https://github.com/dotnet/aspire/issues/9344
